### ISSUE TITLE
CPU load 400%, 500mbit network for 10 seconds during each upgrade or other config change

### ISSUE THREAD

**Issue by GiGurra** [rancher/rancher#10232]
We are having severe issues with cpu and network load when we upgrade any service or change any configuration. Any time a service starts or upgrades (or some metadata state changes), we see our hosts CPU cores being 100% consumed. The issue seems worse the more containers are running on a host - and is especially worrying on our hosts with many containers (50+). Network traffic also goes up dramatically (basically maxes out to several hundred mbits). The issue may be exaggerated on hosts with lots of load balancers.

Rancher metadata service seems to/may be part of the problem after looking through some previous posts about similar issues. This may or may not be related to https://github.com/rancher/rancher/issues/8505.

Our hosts are xeon machines on vmware. Ram usage is not affected.

Image showing the behavior (the normal load state vs spikes during upgrades)
![image](https://user-images.githubusercontent.com/1761299/32089908-fbdb5a1a-baec-11e7-827a-d45c399edf0f.png)

Unfortunately we so far havent been able to pinpoint the exact containers involved because docker stats just gives us the following (with the occasional blip of data, but mostly just dashes) - and rancher just shows 'n/a' on all containers for the host.

```
844b948456d3        --                  -- / --                 --                  --                  --                  --
50aef76a3d0e        --                  -- / --                 --                  --                  --                  --
1f677dc9b6fc        --                  -- / --                 --                  --                  --                  --
7ff6ad33ab8f        --                  -- / --                 --                  --                  --                  --
a578b2035b31        --                  -- / --                 --                  --                  --                  --
9ec6551c5f68        --                  -- / --                 --                  --                  --                  --
```

**All feedback and information on possible workarounds or solutions appreciated!**

During the upgrade rancher metadata logs the following:
```
10/27/2017 8:12:01 AMtime="2017-10-27T06:12:01Z" level=info msg="Update requested for version: 371458"
10/27/2017 8:12:01 AMtime="2017-10-27T06:12:01Z" level=info msg="Downloaded in 334.79348ms"
10/27/2017 8:12:01 AMtime="2017-10-27T06:12:01Z" level=info msg="Generating and reloading answers"
10/27/2017 8:12:01 AMtime="2017-10-27T06:12:01Z" level=info msg="Generating answers"
10/27/2017 8:12:01 AMtime="2017-10-27T06:12:01Z" level=info msg="Generated and reloaded answers"
10/27/2017 8:12:01 AMtime="2017-10-27T06:12:01Z" level=info msg="Applied http://rancher-prod.<obfuscated>/v1/configcontent/metadata-answers?client=v2&requestedVersion=371458?version=371458-4f3c5c96fb170da7fa8781d4ac55192c"
10/27/2017 8:12:01 AMtime="2017-10-27T06:12:01Z" level=info msg="Download and reload in: 410.86856ms"
10/27/2017 8:12:02 AMtime="2017-10-27T06:12:02Z" level=info msg="Update requested for version: 371459"
10/27/2017 8:12:03 AMtime="2017-10-27T06:12:03Z" level=info msg="Downloaded in 308.508509ms"
10/27/2017 8:12:03 AMtime="2017-10-27T06:12:03Z" level=info msg="Generating and reloading answers"
10/27/2017 8:12:03 AMtime="2017-10-27T06:12:03Z" level=info msg="Generating answers"
10/27/2017 8:12:03 AMtime="2017-10-27T06:12:03Z" level=info msg="Generated and reloaded answers"
10/27/2017 8:12:03 AMtime="2017-10-27T06:12:03Z" level=info msg="Applied http://rancher-prod.<obfuscated>/v1/configcontent/metadata-answers?client=v2&requestedVersion=371459?version=371459-4f3c5c96fb170da7fa8781d4ac55192c"
10/27/2017 8:12:03 AMtime="2017-10-27T06:12:03Z" level=info msg="Download and reload in: 735.971854ms"
10/27/2017 8:12:04 AMtime="2017-10-27T06:12:04Z" level=info msg="Update requested for version: 371460"
10/27/2017 8:12:04 AMtime="2017-10-27T06:12:04Z" level=info msg="Downloaded in 499.159892ms"
10/27/2017 8:12:04 AMtime="2017-10-27T06:12:04Z" level=info msg="Generating and reloading answers"
10/27/2017 8:12:05 AMtime="2017-10-27T06:12:05Z" level=info msg="Generating answers"
10/27/2017 8:12:05 AMtime="2017-10-27T06:12:05Z" level=info msg="Generated and reloaded answers"
10/27/2017 8:12:05 AMtime="2017-10-27T06:12:05Z" level=info msg="Applied http://rancher-prod.<obfuscated>/v1/configcontent/metadata-answers?client=v2&requestedVersion=371460?version=371463-4f3c5c96fb170da7fa8781d4ac55192c"
10/27/2017 8:12:05 AMtime="2017-10-27T06:12:05Z" level=info msg="Download and reload in: 1.5487273s"
10/27/2017 8:12:05 AMtime="2017-10-27T06:12:05Z" level=info msg="Update requested for version: 371464"
10/27/2017 8:12:05 AMtime="2017-10-27T06:12:05Z" level=info msg="Downloaded in 27.609869ms"
10/27/2017 8:12:05 AMtime="2017-10-27T06:12:05Z" level=info msg="Generating and reloading answers"
10/27/2017 8:12:06 AMtime="2017-10-27T06:12:06Z" level=info msg="Generating answers"
10/27/2017 8:12:06 AMtime="2017-10-27T06:12:06Z" level=info msg="Generated and reloaded answers"
10/27/2017 8:12:06 AMtime="2017-10-27T06:12:06Z" level=info msg="Applied http://rancher-prod.<obfuscated>/v1/configcontent/metadata-answers?client=v2&requestedVersion=371464?version=371464-4f3c5c96fb170da7fa8781d4ac55192c"
10/27/2017 8:12:06 AMtime="2017-10-27T06:12:06Z" level=info msg="Download and reload in: 565.656088ms"
10/27/2017 8:12:06 AMtime="2017-10-27T06:12:06Z" level=info msg="Update requested for version: 371465"
10/27/2017 8:12:07 AMtime="2017-10-27T06:12:07Z" level=info msg="Downloaded in 407.456029ms"
10/27/2017 8:12:07 AMtime="2017-10-27T06:12:07Z" level=info msg="Generating and reloading answers"
10/27/2017 8:12:07 AMtime="2017-10-27T06:12:07Z" level=info msg="Generating answers"
10/27/2017 8:12:07 AMtime="2017-10-27T06:12:07Z" level=info msg="Generated and reloaded answers"
10/27/2017 8:12:07 AMtime="2017-10-27T06:12:07Z" level=info msg="Applied http://rancher-prod.<obfuscated>/v1/configcontent/metadata-answers?client=v2&requestedVersion=371465?version=371465-4f3c5c96fb170da7fa8781d4ac55192c"
10/27/2017 8:12:07 AMtime="2017-10-27T06:12:07Z" level=info msg="Download and reload in: 946.069244ms"
10/27/2017 8:12:07 AMtime="2017-10-27T06:12:07Z" level=info msg="Update requested for version: 371466"
10/27/2017 8:12:07 AMtime="2017-10-27T06:12:07Z" level=info msg="Downloaded in 77.382536ms"
10/27/2017 8:12:07 AMtime="2017-10-27T06:12:07Z" level=info msg="Generating and reloading answers"
10/27/2017 8:12:08 AMtime="2017-10-27T06:12:08Z" level=info msg="Generating answers"
10/27/2017 8:12:10 AMtime="2017-10-27T06:12:10Z" level=info msg="Generated and reloaded answers"
10/27/2017 8:12:10 AMtime="2017-10-27T06:12:10Z" level=info msg="Applied http://rancher-prod.<obfuscated>/v1/configcontent/metadata-answers?client=v2&requestedVersion=371466?version=371470-4f3c5c96fb170da7fa8781d4ac55192c"
10/27/2017 8:12:10 AMtime="2017-10-27T06:12:10Z" level=info msg="Download and reload in: 2.218432889s"
10/27/2017 8:12:15 AMtime="2017-10-27T06:12:15Z" level=info msg="Update requested for version: 371471"
10/27/2017 8:12:15 AMtime="2017-10-27T06:12:15Z" level=info msg="Downloaded in 252.500025ms"
10/27/2017 8:12:15 AMtime="2017-10-27T06:12:15Z" level=info msg="Generating and reloading answers"
10/27/2017 8:12:15 AMtime="2017-10-27T06:12:15Z" level=info msg="Generating answers"
10/27/2017 8:12:15 AMtime="2017-10-27T06:12:15Z" level=info msg="Generated and reloaded answers"
10/27/2017 8:12:15 AMtime="2017-10-27T06:12:15Z" level=info msg="Applied http://rancher-prod.<obfuscated>/v1/configcontent/metadata-answers?client=v2&requestedVersion=371471?version=371471-4f3c5c96fb170da7fa8781d4ac55192c"
10/27/2017 8:12:15 AMtime="2017-10-27T06:12:15Z" level=info msg="Download and reload in: 346.991149ms"
10/27/2017 8:12:16 AMtime="2017-10-27T06:12:16Z" level=info msg="Update requested for version: 371472"
10/27/2017 8:12:16 AMtime="2017-10-27T06:12:16Z" level=info msg="Downloaded in 225.445044ms"
10/27/2017 8:12:16 AMtime="2017-10-27T06:12:16Z" level=info msg="Generating and reloading answers"
10/27/2017 8:12:17 AMtime="2017-10-27T06:12:17Z" level=info msg="Generating answers"
10/27/2017 8:12:17 AMtime="2017-10-27T06:12:17Z" level=info msg="Generated and reloaded answers"
10/27/2017 8:12:17 AMtime="2017-10-27T06:12:17Z" level=info msg="Applied http://rancher-prod.<obfuscated>/v1/configcontent/metadata-answers?client=v2&requestedVersion=371472?version=371472-4f3c5c96fb170da7fa8781d4ac55192c"
10/27/2017 8:12:17 AMtime="2017-10-27T06:12:17Z" level=info msg="Download and reload in: 1.096078843s"
10/27/2017 8:12:17 AMtime="2017-10-27T06:12:17Z" level=info msg="Update requested for version: 371473"
10/27/2017 8:12:18 AMtime="2017-10-27T06:12:18Z" level=info msg="Downloaded in 51.913304ms"
10/27/2017 8:12:18 AMtime="2017-10-27T06:12:18Z" level=info msg="Generating and reloading answers"
10/27/2017 8:12:18 AMtime="2017-10-27T06:12:18Z" level=info msg="Generating answers"
10/27/2017 8:12:18 AMtime="2017-10-27T06:12:18Z" level=info msg="Generated and reloaded answers"
10/27/2017 8:12:18 AMtime="2017-10-27T06:12:18Z" level=info msg="Applied http://rancher-prod.<obfuscated>/v1/configcontent/metadata-answers?client=v2&requestedVersion=371473?version=371477-4f3c5c96fb170da7fa8781d4ac55192c"
10/27/2017 8:12:18 AMtime="2017-10-27T06:12:18Z" level=info msg="Download and reload in: 984.281581ms"
10/27/2017 8:12:19 AMtime="2017-10-27T06:12:19Z" level=info msg="Update requested for version: 371478"
10/27/2017 8:12:20 AMtime="2017-10-27T06:12:20Z" level=info msg="Downloaded in 465.921958ms"
10/27/2017 8:12:20 AMtime="2017-10-27T06:12:20Z" level=info msg="Generating and reloading answers"
10/27/2017 8:12:21 AMtime="2017-10-27T06:12:21Z" level=info msg="Generating answers"
10/27/2017 8:12:21 AMtime="2017-10-27T06:12:21Z" level=info msg="Generated and reloaded answers"
10/27/2017 8:12:21 AMtime="2017-10-27T06:12:21Z" level=info msg="Applied http://rancher-prod.<obfuscated>/v1/configcontent/metadata-answers?client=v2&requestedVersion=371478?version=371478-4f3c5c96fb170da7fa8781d4ac55192c"
10/27/2017 8:12:21 AMtime="2017-10-27T06:12:21Z" level=info msg="Download and reload in: 1.735553531s"
10/27/2017 8:12:22 AMtime="2017-10-27T06:12:22Z" level=info msg="Update requested for version: 371479"
10/27/2017 8:12:22 AMtime="2017-10-27T06:12:22Z" level=info msg="Downloaded in 649.957085ms"
10/27/2017 8:12:22 AMtime="2017-10-27T06:12:22Z" level=info msg="Generating and reloading answers"
10/27/2017 8:12:23 AMtime="2017-10-27T06:12:23Z" level=info msg="Generating answers"
10/27/2017 8:12:23 AMtime="2017-10-27T06:12:23Z" level=info msg="Generated and reloaded answers"
10/27/2017 8:12:23 AMtime="2017-10-27T06:12:23Z" level=info msg="Applied http://rancher-prod.<obfuscated>/v1/configcontent/metadata-answers?client=v2&requestedVersion=371479?version=371483-4f3c5c96fb170da7fa8781d4ac55192c"
10/27/2017 8:12:23 AMtime="2017-10-27T06:12:23Z" level=info msg="Download and reload in: 1.453227128s"
10/27/2017 8:12:27 AMtime="2017-10-27T06:12:27Z" level=info msg="Update requested for version: 371484"
10/27/2017 8:12:27 AMtime="2017-10-27T06:12:27Z" level=info msg="Downloaded in 217.339852ms"
10/27/2017 8:12:27 AMtime="2017-10-27T06:12:27Z" level=info msg="Generating and reloading answers"
10/27/2017 8:12:27 AMtime="2017-10-27T06:12:27Z" level=info msg="Generating answers"
10/27/2017 8:12:27 AMtime="2017-10-27T06:12:27Z" level=info msg="Generated and reloaded answers"
10/27/2017 8:12:27 AMtime="2017-10-27T06:12:27Z" level=info msg="Applied http://rancher-prod.<obfuscated>/v1/configcontent/metadata-answers?client=v2&requestedVersion=371484?version=371484-4f3c5c96fb170da7fa8781d4ac55192c"
10/27/2017 8:12:27 AMtime="2017-10-27T06:12:27Z" level=info msg="Download and reload in: 345.044459ms"
10/27/2017 8:12:30 AMtime="2017-10-27T06:12:30Z" level=info msg="Update requested for version: 371485"
10/27/2017 8:12:30 AMtime="2017-10-27T06:12:30Z" level=info msg="Downloaded in 458.561189ms"
10/27/2017 8:12:30 AMtime="2017-10-27T06:12:30Z" level=info msg="Generating and reloading answers"
10/27/2017 8:12:30 AMtime="2017-10-27T06:12:30Z" level=info msg="Generating answers"
10/27/2017 8:12:30 AMtime="2017-10-27T06:12:30Z" level=info msg="Generated and reloaded answers"
10/27/2017 8:12:30 AMtime="2017-10-27T06:12:30Z" level=info msg="Applied http://rancher-prod.<obfuscated>/v1/configcontent/metadata-answers?client=v2&requestedVersion=371485?version=371486-4f3c5c96fb170da7fa8781d4ac55192c"
10/27/2017 8:12:30 AMtime="2017-10-27T06:12:30Z" level=info msg="Download and reload in: 534.871133ms"
10/27/2017 8:12:30 AMtime="2017-10-27T06:12:30Z" level=info msg="Update requested for version: 371487"
10/27/2017 8:12:31 AMtime="2017-10-27T06:12:31Z" level=info msg="Downloaded in 284.11857ms"
10/27/2017 8:12:31 AMtime="2017-10-27T06:12:31Z" level=info msg="Generating and reloading answers"
10/27/2017 8:12:32 AMtime="2017-10-27T06:12:32Z" level=info msg="Generating answers"
10/27/2017 8:12:32 AMtime="2017-10-27T06:12:32Z" level=info msg="Generated and reloaded answers"
10/27/2017 8:12:32 AMtime="2017-10-27T06:12:32Z" level=info msg="Applied http://rancher-prod.<obfuscated>/v1/configcontent/metadata-answers?client=v2&requestedVersion=371487?version=371488-4f3c5c96fb170da7fa8781d4ac55192c"
10/27/2017 8:12:32 AMtime="2017-10-27T06:12:32Z" level=info msg="Download and reload in: 2.032476785s"
```

-----------------------

**Rancher versions:**
rancher/server: 1.6.7
rancher/agent: 1.2.5

**Infrastructure Stack versions:**
healthcheck: 0.3.1
ipsec: 0.11.7
network-services:
  metadata:v0.9.3
  network-manager:v0.7.7
scheduler: 0.8.2
kubernetes (if applicable): -

**Docker version: (`docker version`,`docker info` preferred)**
Docker version 17.03.1-ce, build c6d412e

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
Linux rancher-slave-prod01 4.4.0-81-generic #104-Ubuntu SMP Wed Jun 14 08:17:06 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
VMware

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single rancher server
5 host
internal db

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Comment by aemneina** [comment]
seems to be a dupe of:
https://github.com/rancher/rancher/issues/9957
https://github.com/rancher/rancher/issues/9373
https://github.com/rancher/rancher/issues/8505

There are a couple fixes in progress, one is to throttle metadata requests, another is to impose CPU quotas.

**Comment by aemneina** [comment]
The fix should ship with 1.6.11. I'll close this out. Feel free to comment if you think I'm missing something here, and I can reopen.
